File: Others/007.Reverse-Integer/007.Reverse-Integer.cpp
class Solution {
public:
    int reverse(int x) 
    {
        if (x==INT_MIN) return 0;
        
        int sign = 1;
        if (x<0) sign = -1;
        
        x = abs(x); 
        
        int ret = 0;
        while (x>0)
        {
            if (ret > INT_MAX/10) return 0;
            if (ret*10 > INT_MAX - x%10 ) return 0;
            
            ret = ret*10 + x%10;
            x = x/10;
        }
        
        return ret*sign;
    }
};

File: Math/007.Reverse-Integer.py

class Solution:
    def reverse(self, x: int) -> int:
        if x < 0:
            signo = -1
        else:
            signo = 1
        x = abs(x)

        resultado = 0

        while x > 0:
            digito = x % 10  #last digit of x
            resultado = (resultado * 10 + digito) #add digit to resultado
            x = x // 10
        resultado = resultado * signo

        if resultado < -2_147_483_648 or resultado > 2_147_483_647:
            return 0
        else:
            return resultado
